### PR TITLE
fix(parental-leave): Fixes to duration slider screen

### PR DESCRIPTION
**Pull request by madebynoam**
This PR fixes 2 issues and updates 1 value:
1. Allow the user to move the duration slider beyond 6 months
2. Resolve an error if the user refreshes while on the duration slider screen.
3. Change the maxMonths value to 24

Notes:
On the duration slider screen, we can't determine if the user will spread their leave beyond 6 months with a percent ratio lower than 100. So we will take out the check `"if the endDate is inside the allowed range of days/months"` for now.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/core/src/lib/formUtils.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export function getValueViaPath(
 ): unknown | undefined {
   // Errors from dataSchema with array of object looks like e.g. `{ 'periods[1].startDate': 'error message' }`
   if (path.match(/.\[\d\]\../g) && !containsArray(obj)) {
-    return obj?.[path]
+    return obj?.[path] || defaultValue
```
Comment (jeremybarbet) [inline libs/application/core/src/lib/formUtils.ts:44]:
```suggestion
    return obj?.[path] ?? defaultValue
```

**Review by jeremybarbet — COMMENTED** [review]
Just a minor comment

File: libs/application/core/src/lib/formUtils.ts (hunk 1 above)
Comment (madebynoam) [reply libs/application/core/src/lib/formUtils.ts]: Nice!

**Review by jeremybarbet — APPROVED** [review]
LGTM

**Pull request merged**